packages/salesforcedx-vscode-core/src/commands/forceAuthWebLogin.ts

```diff
@@ -29,14 +41,72 @@ class ForceAuthWebLoginExecutor extends SfdxCommandletExecutor<{}> {
   }
 }
 
+class ForceAuthWebDemoModeLoginExecutor extends SfdxCommandletExecutor<{}> {
+  public build(data: {}): Command {
+    return new SfdxCommandBuilder()
+      .withDescription(
+        nls.localize('force_auth_web_login_authorize_dev_hub_text')
+      )
+      .withArg('force:auth:web:login')
+      .withArg('--noprompt')
+      .withArg('--json')
+      .withArg('--setdefaultdevhubusername')
+      .build();
+  }
+
+  public async execute(response: ContinueResponse<{}>): Promise<void> {
+    const cancellationTokenSource = new CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+
+    const execution = new CliCommandExecutor(this.build({}), {
+      cwd: workspace.rootPath
+    }).execute(cancellationToken);
+
+    notificationService.reportExecutionError(
+      execution.command.toString(),
+      (execution.stderrSubject as any) as Observable<Error | undefined>
+    );
+
+    channelService.streamCommandOutput(execution);
+    CancellableStatusBar.show(execution, cancellationTokenSource);
+    taskViewService.addCommandExecution(execution, cancellationTokenSource);
+
+    const cmdOutput = new CommandOutput();
+    const result = await cmdOutput.getCmdResult(execution);
+
+    if (isTrialOrg(JSON.parse(result))) {
```

Comment:
Reading this is kind of confusing since it reads like you're prompting for log out if it's a trial org but isTrialOrg returns false if it is a trial org.

Reply:
Yup, I was tripped out by it too. I renamed it to isProdOrg instead in my latest commit.